test/lib/command/command.test.js

```diff
@@ -813,7 +1492,140 @@ describe('/command', function () {
+        setTimeout(() => {
+          expect(response.message).to.equal(errorMessage);
+          done();
+        }, 1);
+      });
+
+      testBots.start().then((botEvt) => {
+        botEvt[0].on('message', onMessageSpy);
+
+        botEvt[0].on('connect', () => {
+          botEvt[0].injectMessage(slackMessage);
+        });
+      });
+    });
+
+    it('Should call alertTask for alert command', function (done) {
+      slackMessage.text = 'alert';
+      delete errorContext.error;
+      errorContext.noOfErrors = 1;
+      errorContext.failedParams = [{ error: 'err!! you are missing ' +
+          'another argument' }];
+      errorContext.parsedMessage = messageParser(slackMessage);
+      const errorMessage = responseHandler.generateErrorTemplate('testbot1',
+        testBots.bots[0].config.botCommand, errorContext);
+
+      const onMessageSpy = sandbox.spy((response) => {
+        setTimeout(() => {
+          expect(response.message).to.equal(errorMessage);
+          done();
+        }, 1);
+      });
+
+      testBots.start().then((botEvt) => {
+        botEvt[0].on('message', onMessageSpy);
+
+        botEvt[0].on('connect', () => {
+          botEvt[0].injectMessage(slackMessage);
+        });
+      });
+    });
+
+    it('Should call schedule command', function (done) {
+      sandbox.stub(storage, 'getEvents').callsFake(() => {
+        return Promise.resolve({
+          events: {},
+          schedule: config.schedule,
+        });
+      });
+      const clock = sandbox.useFakeTimers(new Date());
+
+      const onMessageSpy = sandbox.spy((response) => {
+        if (onMessageSpy.getCalls().length === 1) {
+          expect(response.message).to.eq('Hello 1');
+          clock.tick(60000);
+        }
+        if (onMessageSpy.getCalls().length === 2) {
+          expect(response.message).to.eq('Hello 1');
+          done();
+        }
+      });
+      testBots.start().then((botEvt) => {
+        botEvt[0].on('message', onMessageSpy);
+
+        botEvt[0].on('connect', () => {
+          clock.tick(60000);
+        });
+      });
+    });
+  });
+
+  describe('Test command types - data function returns Promise', function () {
+    beforeEach(function () {
+      initTestSetup({
+        config: config.dataPromise.commandTypeBots,
+        slackMessage: {
+          channel: 'D0GL06JD7',
+          user: 'U0GG92T45',
+          text: 'ping 1',
+        },
+      });
+    });
+
+    it('Should call getData for data command', function (done) {
+      slackMessage.text = 'ping';
+
+      const onMessageSpy = sandbox.spy((response) => {
+        setTimeout(() => {
+          expect(response.message).to.equal('Hello 1');
+          done();
+        }, 1);
+      });
+
+      testBots.start().then((botEvt) => {
+        botEvt[0].on('message', onMessageSpy);
+
+        botEvt[0].on('connect', () => {
+          botEvt[0].injectMessage(slackMessage);
+        });
+      });
+    });
+
```

Comment:
This test has confounded me :(

Reply:
Max, the change you made in data.js needed here as well https://github.com/usubram/slack-quick-bots/blob/master/lib/command/commands/recursive.js#L70 for this test to pass.

Make sense to extract into a common function in https://github.com/usubram/slack-quick-bots/blob/master/lib/command/command.js

Reply:
Done, test re-enabled as well. I'll leave the extraction to you since you plan on doing a larger refactor in this codebase.